data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/SqsService.java

```diff
@@ -23,22 +24,25 @@ public class SqsService {
     private final S3Service s3Accessor;
     private final SqsClient sqsClient;
     private final PluginMetrics pluginMetrics;
+    private final AcknowledgementSetManager acknowledgementSetManager;
 
     private Thread sqsWorkerThread;
 
-    public SqsService(final S3SourceConfig s3SourceConfig,
+    public SqsService(final AcknowledgementSetManager acknowledgementSetManager,
+                      final S3SourceConfig s3SourceConfig,
                       final S3Service s3Accessor,
                       final PluginMetrics pluginMetrics) {
         this.s3SourceConfig = s3SourceConfig;
         this.s3Accessor = s3Accessor;
         this.pluginMetrics = pluginMetrics;
+        this.acknowledgementSetManager = acknowledgementSetManager;
         this.sqsClient = createSqsClient();
     }
 
     public void start() {
         final Backoff backoff = Backoff.exponential(INITIAL_DELAY, MAXIMUM_DELAY).withJitter(JITTER_RATE)
                 .withMaxAttempts(Integer.MAX_VALUE);
-        sqsWorkerThread = new Thread(new SqsWorker(sqsClient, s3Accessor, s3SourceConfig, pluginMetrics, backoff));
+        sqsWorkerThread = new Thread(new SqsWorker(acknowledgementSetManager, sqsClient, s3Accessor, s3SourceConfig, pluginMetrics, backoff));
```

Comment:
I think this will work for now.

But, I also think once we have scan that the acknowledgement set should be handled by the S3 Object/Select workers. Then we can have a better mechanism for them to callback to the either the SQS or scan handlers.

Reply:
I think Scan can exist without SQS or with SQS. So, I think it needs to be handled separately anyways and I don't think that would change SQS related acknowledgements.